otherlibs/stdune/src/string.ml

```diff
@@ -328,3 +328,7 @@ let contains_double_underscore =
 ;;
 
 let last s = if length s > 0 then Some s.[length s - 1] else None
+
+let replace_char s ~from ~to_ =
+  split_on_char s ~sep:from |> String.concat ~sep:(String.make 1 to_)
+;;
```

Comment:
I haven't benchmarked this but I feel it would be easier to use `String.map` as it will probably allocate dramatically fewer strings.

Something like

```ocaml
let replace_char ~from ~to_ =
  String.map ~f:(fun c ->
    if Char.equal c from then to_ else c)
```

Reply:
Ah good idea. I didn't know about `String.map`.